### PR TITLE
fixes thermal pistols doing literally nothing on impact

### LINKED ISSUE #79167: Thermal Pistols don't hurt on point blanks, cryo pistol just seems to do nothing
Reporting client version: 514.1589

## Round ID:
[217267](https://scrubby.melonmesa.com/round/217267)

## Testmerges:
- [Spider Infestation Balance Rebalance Expansion Part 2](https://www.github.com/tgstation/tgstation/pull/78705)
- [Adds a subtle ghost poll](https://www.github.com/tgstation/tgstation/pull/79105)

## Reproduction:
1) Get a thermal and cryo pistol.
2) Fire at someone multiple times with an inferno pistol. For me, it did do damage at first and made me feel like an idiot.
3) Fire at someone with a cryo pistol. This should just do nothing.
4) If those do not work, try point-blanking a target with combat mode off. It should look like it hit them while actually doing no damage.
5) Repeat with thermal and cryo pistols and it should do nothing outright.

## Notes:
First observed on previous sybil round by attempting to shoot a monkey with either pistols, doing no damage. After, I had someone shoot me with a cryo pistol and took no damage.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Fixes https://github.com/tgstation/tgstation/issues/79167

## Why It's Good For The Game

refactors sometimes overlook things

## Changelog
:cl:
fix: The nanites inside of thermal pistols are once again angry, and aggressively want to burn/puncture people.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile/energy/thermal.dm [hunk 1]
```diff
@@ -10,10 +10,10 @@
 	bare_wound_bonus = 10
 	impact_effect_type = /obj/effect/temp_visual/impact_effect/red_laser
 
-/obj/projectile/energy/inferno/on_hit(atom/target, blocked, pierce_hit)
-	..()
+/obj/projectile/energy/inferno/on_hit(atom/target, blocked = 0, pierce_hit)
+	. = ..()
 	if(!ishuman(target))
-		return
+		return BULLET_ACT_HIT
```
Comment (Fikou) [inline code/modules/projectiles/projectile/energy/thermal.dm:16]: are we sure we want to override the return value?

Comment (necromanceranne) [reply code/modules/projectiles/projectile/energy/thermal.dm:16]: I'm not as sure with this one lemme revert that particular change.

File: code/modules/projectiles/projectile/energy/thermal.dm [hunk 2]
```diff
@@ -10,10 +10,10 @@
 	bare_wound_bonus = 10
 	impact_effect_type = /obj/effect/temp_visual/impact_effect/red_laser
 
-/obj/projectile/energy/inferno/on_hit(atom/target, blocked, pierce_hit)
-	..()
+/obj/projectile/energy/inferno/on_hit(atom/target, blocked = 0, pierce_hit)
+	. = ..()
 	if(!ishuman(target))
-		return
+		return ..()
```
Comment (Jacquerel) [inline code/modules/projectiles/projectile/energy/thermal.dm:16]: Now hold on I think you want to change _either_ of these lines but not _both_
Now you're calling the parent proc twice in both of these instances

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile/energy/thermal.dm (hunk 2 above)
Comment (necromanceranne) [reply code/modules/projectiles/projectile/energy/thermal.dm:16]: Ah I think you're right sorry it's late for me

**Pull request merged**